sf_stats.Api/sf_stats.Api/Controllers/DivisionController.cs

```diff
@@ -0,0 +1,133 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using sf_stats.Api.Interfaces;
+using sf_stats.Domain.DomainObjects;
+using sf_stats.Domain.Dtos;
+using sf_stats.Domain.Entities;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace sf_stats.Api.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class DivisionController : ControllerBase
+    {
+        private readonly ILogger<LogController> _logger;
+        private readonly IDivisionService _divisionService;
+        private readonly IMapper _mapper;
+
+        public DivisionController(ILogger<LogController> logger, IMapper mapper, IDivisionService divisionService)
+        {
+            _logger = logger;
+            _mapper = mapper;
+            _divisionService = divisionService;
+        }
+
+        /// <summary>
+        /// Get Divisions based on the filter provided
+        /// </summary>
+        /// <remarks>
+        /// Sample request:
+        ///
+        ///     GET /GetDivisions
+        ///     {
+        ///        "name": "Test Division",
+        ///        "code": "TTD"
+        ///     }
+        ///
+        /// </remarks>
+        /// <param name="name">Division name</param>
+        /// <param name="code">Division code name</param>
+        /// <returns> Division entries based on the filter provided</returns>
+        [HttpGet]
+        [Produces("application/json")]
+        public async Task<ActionResult<IEnumerable<DivisionDto>>> GetDivisions(
+            [FromQuery] string name,
+            [FromQuery] string code,
+            [FromQuery] int? id)
+        {
```

Comment:
Looks like a build warning? Don't think it breaks anything but mind adding param?

Reply:
updated